frontend/lib/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -63,6 +92,55 @@ interface State {
   error?: Error
 }
 
+/**
+ * Prepares the vega-lite spec for selections by transforming the select parameters
+ * to a full object specification and by automatically adding encodings (if missing)
+ * to point selections.
+ *
+ * The changes are applied in-place to the spec object.
+ *
+ * @param spec The Vega-Lite specification of the chart.
+ */
+export function prepareSpecForSelections(spec: any): void {
+  if ("params" in spec && "encoding" in spec) {
+    spec.params.forEach((param: any) => {
+      if (!("select" in param)) {
+        // We are only interested in transforming select parameters.
+        // Other parameters are skipped.
+        return
+      }
+
+      if (["interval", "point"].includes(param.select)) {
+        // The select object can be either a single string (short-hand) specifying
+        // "interval" or "point" or an object that can contain additional
+        // properties as defined here: https://vega.github.io/vega-lite/docs/selection.html
+        // To make our life easier, we convert the short-hand notation to the full object specification.
+        param.select = {
+          type: param.select,
+        }
+      }
+
+      if (!("type" in param.select)) {
+        // The type property is required in the spec.
+        // But we check anyways and skip all parameters that don't have it.
+        return
+      }
```

Comment:
Couldn't we check this directly in the beginning of the loop where we also check the existence of the `select` key?

Reply:
Nope, since the shorthand notation above is a valid usage, but it doesn't yet have the `type` attribute. That's why we unwrap the short-hand notation.